### PR TITLE
Optionally parse non-required well-known token metadata properties from server

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

ADP-413

# Overview

- [ ] I have added the missing parser for all well-known properties.

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Primitive/Types/TokenPolicy.hs [hunk 1]
```diff
@@ -126,12 +131,29 @@ instance FromText TokenName where
...
-    -- , acronym :: Text            -- TODO: needs metadata-server support
     , description :: Text
-    -- , url :: Text                -- TODO: needs metadata-server support
-    -- , logoBase64 :: ByteString   -- TODO: needs metadata-server support
-    -- , unit :: AssetUnit          -- TODO: needs metadata-server support
+    , acronym :: Maybe Text
+    , url :: Maybe Text
+    , logo :: Maybe AssetLogo
+    , unit :: Maybe AssetUnit
     } deriving stock (Eq, Ord, Generic)
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Primitive/Types/TokenPolicy.hs:139]: 👌

**Review by Anviking — APPROVED** [review]
lgtm

File: lib/core/test/data/Cardano/Wallet/TokenMetadata/golden1.json [hunk 2]
```diff
@@ -72,7 +72,10 @@
         "subject": "extra fields",
         "name": { "value": "Token2" },
         "description": { "value": "description2" },
-        "acronym": { "value": "acronym2" }
+        "acronym": { "value": "acronym2" },
+        "unit": { "value": { "decimals": 14, "name": "unit2" } },
+        "url": { "value": "https://iohk.io" },
+        "logo": { "value": "QWxtb3N0IGEgbG9nbw==" }
```
Comment (Anviking) [inline lib/core/test/data/Cardano/Wallet/TokenMetadata/golden1.json:78]: We should probably add data + checks for these new fields in the integration tests too

Comment (KtorZ) [reply lib/core/test/data/Cardano/Wallet/TokenMetadata/golden1.json:78]: Yeah. If you know precisely how, feel free :sweat_smile:

**Comment by KtorZ** [comment]
bors retry

**Comment by KtorZ** [comment]
bors retry

please.

**Comment by rvl** [comment]
bors r+

please

**Pull request merged**